core/src/main/java/com/gentics/mesh/core/endpoint/node/BinaryUploadHandlerImpl.java

```diff
@@ -214,9 +216,12 @@ public void handleUpdateField(InternalActionContext ac, String nodeUuid, String
 				if (log.isDebugEnabled()) {
 					log.debug("Error detected. Purging previously stored upload for tempId {}", tmpId, e);
 				}
-				return binaryStorage.purgeTemporaryUpload(tmpId).doOnError(e1 -> {
-					log.error("Error while purging temporary upload for tempId {}", tmpId, e1);
-				}).onErrorComplete().andThen(Single.error(e));
+				return binaryStorage.purgeTemporaryUpload(tmpId)
+						// The file has already been deleted = not an error
```

Comment:
The reason for playing with tests is that after the fix the tests run **actually** asynchronous, and cause races in the handler.